addons/luamodel/luatypes.h

```diff
@@ -0,0 +1,385 @@
+               );
+      break;
+    //default:
+    //  throw RigidBodyDynamics::Errors::RBDLFileParseError(
+    //        "Invalid number of DOFs for joint.");
+    //  break;
+    }
+  }
+
+  stackRestore();
+
+  return result;
+}
+
+
+//==============================================================================
+template<>
+RigidBodyDynamics::Body
+LuaTableNode::getDefault<RigidBodyDynamics::Body>(
+    const RigidBodyDynamics::Body &default_value)
+{
+  RigidBodyDynamics::Body result = default_value;
+
+  if (stackQueryValue()) {
+    LuaTable vector_table = LuaTable::fromLuaState (luaTable->L);
+
+    double mass = 0.;
+    RigidBodyDynamics::Math::Vector3d com(
+          RigidBodyDynamics::Math::Vector3d::Zero(3));
+    RigidBodyDynamics::Math::Matrix3d inertia(
+          RigidBodyDynamics::Math::Matrix3d::Identity(3,3));
+
+    mass = vector_table["mass"];
+    com = vector_table["com"]
+        .getDefault<RigidBodyDynamics::Math::Vector3d>(com);
+    inertia = vector_table["inertia"]
+        .getDefault<RigidBodyDynamics::Math::Matrix3d>(inertia);
+
+    result = RigidBodyDynamics::Body (mass, com, inertia);
+  }
+
+  stackRestore();
+
+  return result;
+}
+
+//==============================================================================
+
+template<> Point
+LuaTableNode::getDefault<Point>(
+  const Point &default_value)
+{ 
+  Point result = default_value;
+
+  if (stackQueryValue()) {
+    LuaTable point_table = LuaTable::fromLuaState (luaTable->L);
+
+    result.name         = point_table["name"].get<std::string>();
+    result.point_local  = point_table["point"];
+    result.body_name    = point_table["body"].get<std::string>();
+  }
+
+  stackRestore();
+
+  return result;
+}
+
+//==============================================================================
+
+template<> MotionCaptureMarker
+LuaTableNode::getDefault<MotionCaptureMarker>(
+  const MotionCaptureMarker &default_value)
+{
+  MotionCaptureMarker result = default_value;
+
+  if (stackQueryValue()) {
+    LuaTable marker_table = LuaTable::fromLuaState (luaTable->L);
+
+    result.name         = marker_table["name"].get<std::string>();
+    result.point_local  = marker_table["point"]
+                            .getDefault<RigidBodyDynamics::Math::Vector3d>(
+                              result.point_local);
+  }
+
+  stackRestore();
+
+  return result;
+}
+
+//==============================================================================
+
+template<> LocalFrame
+LuaTableNode::getDefault<LocalFrame>(
+  const LocalFrame &default_value)
+{
+  LocalFrame result = default_value;
+
+  if (stackQueryValue()) {
+    LuaTable local_frame_table = LuaTable::fromLuaState (luaTable->L);
+
+    result.name      = local_frame_table["name"].get<std::string>();
+    result.body_name = local_frame_table["body"].get<std::string>();
+    result.r = local_frame_table["r"]
+                .getDefault<RigidBodyDynamics::Math::Vector3d>(result.r);
```

Comment:
If in Lua `frame.r` is not given then `result.r` will be undefined. The `getDefault()` function assumes that you provide the value that should be used if the value is not found in the Lua table. In this case you likely want
```
.getDefault<RigidBodyDynamics::Math::Vector3d>(Vector3d::Zero());
```
this also holds for `result.E` below where you need to use `Matrix3d::Identity()` as default value.

Reply:
I've made this change. However, I think the problem might have been my understanding. Is this correct?

- When the line 'LocalFrame result = default_value' is executed the default constructor for LocalFrame is called and thus the r and E fields have values.
- By passing result.r and result.E into the 'getDefault' command the values for r & E set in the default constructor are used (in this case a Vector3d and Matrix3d of nans).

Does this not work?